a10_neutron_lbaas/plumbing_hooks.py

```diff
@@ -21,11 +24,33 @@ def __init__(self, driver):
         self.driver = driver
         self.appliance_hash = acos_client.Hash(self.driver.config.devices.keys())
 
-    def select_device(self, tenant_id):
+    def select_device_hash(self, tenant_id):
         # Must return device dict from config.py
         s = self.appliance_hash.get_server(tenant_id)
         return self.driver.config.devices[s]
 
+    def select_device_db(self, tenant_id):
+        db = db_api.get_session()
+
+        # See if we have a saved tenant
+        a10 = db.query(models.A10TenantBinding).filter(
+            models.A10TenantBinding.tenant_id == tenant_id).one_or_none()
+        if a10 is not None and a10.device_name in self.driver.config.devices:
+            return self.driver.config.devices[a10.device_name]
```

Comment:
If the config's not in config.devices, this needs to error rather than continuing. If it continues, you assign the user to a new device and `.one_or_none()` fails on all future calls to `select_device_db`.